### PR TITLE
Better save of agents

### PR DESCRIPTION
**Pull request by CorentinBrtx**

### CODE REVIEW

File: game/episode.py [hunk 1]
```diff
@@ -88,3 +91,32 @@ def run_episode_against_self(
...
+
+def watch_game(agent1: Union[str, Agent], agent2: Union[str, Agent]):
+    """Watch a game between two saved agents"""
+
+    env = ConnectFourGymEnv()
+
+    agent1_path = None
+    agent2_path = None
+
+    if isinstance(agent1, str):
```
Comment (Castavo) [inline game/episode.py:104]: Pas mal de redondance quand même

File: game/training/train.py [hunk 2]
```diff
@@ -46,19 +48,38 @@ def evaluate_agent(
...
 
+    architecture = list(map(str, next(agent.value_network.children())))
+
+    config = {
+        "n_episodes": 0,
+        "discount": discount,
+        "epsilon": agent.epsilon,
+        "value_network": architecture,
+        "learning_rate": agent.learning_rate,
+    }
```
Comment (Castavo) [inline game/training/train.py:81]: We should also add this to the train not against self function, but we should be smart and find a way to somehow factorize it

**Review by Castavo — APPROVED** [review]

File: game/episode.py [hunk 3]
```diff
@@ -88,3 +91,32 @@ def run_episode_against_self(
     rewards[-2] = env.get_final_reward(before_last_player)
 
     return states[::2], states[1::2], rewards[::2], rewards[1::2]
+
+
+def watch_game(agent1: Union[str, Agent], agent2: Union[str, Agent]):
+    """Watch a game between two saved agents"""
```
Comment (maximecharpentierdata) [inline game/episode.py:97]: Pas seulement saved du coup, on peut directement loader un agent non ?

**Review by maximecharpentierdata — APPROVED** [review]
C'est bo

**Review by maximecharpentierdata — APPROVED** [review]

**Pull request merged**